src/go/rpk/pkg/cli/cluster/config/get.go

```diff
@@ -69,3 +86,34 @@ output, use the 'edit' and 'export' commands respectively.`,
 
 	return cmd
 }
+
+func getCloudConfig(ctx context.Context, cfg *config.Config, p *config.RpkProfile, configName string) (string, error) {
+	cloudClient := publicapi.NewCloudClientSet(cfg.DevOverrides().PublicAPIURL, p.CurrentAuth().AuthToken)
+	req := connect.NewRequest(&controlplanev1.GetClusterRequest{
+		Id: p.CloudCluster.ClusterID,
+	})
+
+	cluster, err := cloudClient.Cluster.GetCluster(ctx, req)
+	if err != nil {
+		var ce *connect.Error
+		if errors.As(err, &ce) {
+			if ce.Code() == connect.CodePermissionDenied {
+				return "", fmt.Errorf("this user does not have permission to get cluster information, please check your role or contact admin")
+			}
+			if ce.Code() == connect.CodeNotFound {
+				return "", fmt.Errorf("cluster not found. Please ensure the cluster exists in the cloud")
+			}
+		}
+		return "", fmt.Errorf("internal error while getting redpanda cloud configs: %v", err)
+	}
+	configs := cluster.Msg.GetCluster().GetClusterConfiguration().GetComputedProperties()
+	value, ok := configs.GetFields()[configName]
+	if !ok {
+		return "", fmt.Errorf("unknown property: %s. \nsome configurations are not available for retrieval in a cloud cluster", configName)
+	}
+	if value == nil {
+		return "", fmt.Errorf("unknown property: %s. \nsome configurations are not available for retrieval in a cloud cluster", configName)
+	}
```

Comment:
Is the same error message for 2 different scenarios. Is there a way to tell if the requested configuration is not available to be read?

There are some nullable properties in our configuration (around ~70 configs, [for example](https://docs.redpanda.com/current/reference/properties/cluster-properties/#audit_excluded_principals)). Doesn't the `value==nil` is a valid value in some of them?

Reply:
I agree that validation is not necessary